certbot/tests/main_test.py

```diff
@@ -1404,6 +1404,40 @@ def test_update_account_no_existing_accounts(self):
                  "user@example.org"])
             self.assertTrue("Could not find an existing account" in x[0])
 
+    def test_update_account_remove_email(self):
+        with mock.patch('certbot._internal.eff.prepare_subscription') as mock_prepare:
+            with mock.patch('certbot._internal.main._determine_account') as mocked_det:
+                with mock.patch('certbot._internal.main.account') as mocked_account_module:
+                    with mock.patch('certbot._internal.main.client') as mocked_client:
```

Comment:
If mocking for the entire span of the method, [doing it like this](https://github.com/certbot/certbot/blob/860af81fef3042134b2b71a6b71b4240c75b1886/certbot/tests/main_test.py#L149-L158) can save you from excessive nesting.

Reply:
I had used [test_update_account_with_email](https://github.com/certbot/certbot/blob/860af81fef3042134b2b71a6b71b4240c75b1886/certbot/tests/main_test.py#L1407-L1435) as a template to learn how tests were done and wasn't sure about decorator / with block preferences. Happy to refactor to use all decorators!